File: Data Structures & Algorithms/max-area-of-island/submission-0.java
class Solution {
    public int maxAreaOfIsland(int[][] grid) {

        int rows = grid.length; int cols = grid[0].length;
        int maxIsland = 0;

        for (int r = 0; r < rows; r++) {
            for (int c = 0; c < cols; c++) {
                if (grid[r][c] == 1) {
                    maxIsland = Math.max(maxIsland, dfs(grid, r, c));
                }
            }
        }

        return maxIsland;
        
    }

    private int dfs(int[][] grid, int r, int c) {

        if (r < 0 || r >= grid.length || c < 0 || c >= grid[0].length || grid[r][c] == 0) {
            return 0;
        } else {
            grid[r][c] = 0;
            int res = 1;
            res += dfs(grid, r + 1, c);
            res += dfs(grid, r, c + 1);
            res += dfs(grid, r - 1, c);
            res += dfs(grid, r, c - 1);
            return res;
        }

        
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cs
public class Solution {
    public int MaxProfit(int[] prices) {

        int res = 0;
        int current = prices[prices.Length - 1];
        int max = 0;

        for (int i = prices.Length - 2; i >= 0; i--) {
            if (prices[i] < prices[i + 1]) {
                int tmp = current - prices[i];
                max = Math.Max(max, tmp);
            } else {
                res += max;
                max = 0;
                current = prices[i];
            }
        }
        res += max;
        return res;
        
    }
}